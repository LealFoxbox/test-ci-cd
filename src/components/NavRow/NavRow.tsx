import React from 'react';
import { View } from 'react-native';
import { Card, Text, useTheme } from 'react-native-paper';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import { styled } from 'src/paperTheme';
import { getAccessibilityAndAutomationProps } from 'src/utils/accessibility';

const Container = styled.TouchableOpacity`
  background-color: white;
  padding: 10px;
`;

interface RowProps {
  label: string;
  content?: React.ReactNode;
  accessibilityLabel?: string;
  icon?: string;
  onPress: () => void;
}

const NavRow: React.FC<RowProps> = ({ accessibilityLabel, label, content, icon, onPress }) => {
  const theme = useTheme();

  return (
    <Container
      onPress={onPress}
      accessibilityRole="button"
      {...getAccessibilityAndAutomationProps(accessibilityLabel || label)}
    >
      <View style={{ flexDirection: 'row', alignItems: 'center', width: '100%', paddingRight: 15, paddingLeft: 5 }}>
        {icon && (
          <MaterialCommunityIcons name={icon} size={24} color={theme.colors.primary} style={{ marginLeft: 15 }} />
        )}
        <Card.Content style={{ flex: 1 }}>
          <Text>{label}</Text>
          {content || null}
        </Card.Content>
        <MaterialIcons name="chevron-right" size={24} color={theme.colors.placeholder} />
      </View>
    </Container>
  );
};

export default NavRow;
