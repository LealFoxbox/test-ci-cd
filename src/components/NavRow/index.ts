export * from './NavRow';
export { default } from './NavRow';
