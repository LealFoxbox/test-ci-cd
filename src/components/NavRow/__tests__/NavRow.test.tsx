import React from 'react';
import { render } from '@testing-library/react-native';

import NavRow from '../NavRow';

describe('NavRow', () => {
  it('should render', () => {
    const result = render(<NavRow label="label" icon="chevron-left" onPress={() => undefined} />);

    expect(result).toMatchSnapshot();
  });
});
