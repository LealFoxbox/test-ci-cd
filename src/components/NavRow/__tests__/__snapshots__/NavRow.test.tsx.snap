// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavRow should render 1`] = `
<View
  accessibilityLabel="label"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
  testID="label"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": 5,
        "paddingRight": 15,
        "width": "100%",
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 24,
          },
          Object {
            "marginLeft": 15,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰅁
    </Text>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          undefined,
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        label
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
